src/dmd/dtoh.d

```diff
@@ -0,0 +1,1724 @@
+/**
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language).
+ *
+ * Copyright:   Copyright (C) 1999-2019 by The D Language Foundation, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:      $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/dtohd, _dtoh.d)
+ * Documentation:  https://dlang.org/phobos/dmd_dtoh.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/dtoh.d
+ */
+module dmd.dtoh;
+
+import core.stdc.stdio;
+import core.stdc.string;
+import core.stdc.ctype;
+
+import dmd.astcodegen;
+import dmd.arraytypes;
+import dmd.globals;
+import dmd.identifier;
+import dmd.json;
+import dmd.mars;
+import dmd.root.array;
+import dmd.root.file;
+import dmd.root.filename;
+import dmd.root.rmem;
+import dmd.visitor;
+import dmd.tokens;
+
+import dmd.root.outbuffer;
+import dmd.utils;
+
+//debug = Debug_DtoH;
+
+private struct DMDType
+{
+    __gshared static Identifier c_long;
+    __gshared static Identifier c_ulong;
+    __gshared static Identifier c_longlong;
+    __gshared static Identifier c_ulonglong;
+    __gshared static Identifier c_long_double;
+    //version(BUILD_COMPILER)
```

Comment:
Its probably best to remove all the commented out version(BUILD_COMPILER) and put it behind a (hidden) compiler switch instead.

Reply:
AFAICT the purpose of this spearate program was to only generate header files without actual codegen?

Assuming this is the case one could achieve the same behaviour using `-c -o-`(?)

Reply:
No the purpose of that program was to do all the weird stuff that the DMD headers need see `isVisitorClass`, `isIgnoredModule` and friends. And separate it from the behaviour that compiler users would expect from the `-HC` switch.

Reply:
Thanks. I've hidden this functionality behind an enum `isBuildingCompiler` until we define an appropriate name for this  flag. 

But the following code snippet got me thinking whether special casing for DMD is a good idea or if there's a deeper problem with the current header generation.

https://github.com/dlang/dmd/blob/3585ac12f887d90c7f621c1b65d7af72a46bd8db/src/dmd/dtoh.d#L1294-L1329